### PR TITLE
Fix MessageFormatter for PHP7

### PR DESCRIPTION
**Pull request by subdee**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #11528 |

### CODE REVIEW

**Comment by SilverFire** [comment]
We need a changelog line and a test case to verify this behavior. Could you do it, please?

File: framework/i18n/MessageFormatter.php [hunk 1]
```diff
@@ -99,10 +99,14 @@ public function format($pattern, $params, $language)
...
-        $formatter = new \MessageFormatter($language, $pattern);
-        if ($formatter === null) {
-            $this->_errorCode = intl_get_error_code();
-            $this->_errorMessage = 'Message pattern is invalid: ' . intl_get_error_message();
+        try {
+            $formatter = new \MessageFormatter($language, $pattern);
+            if ($formatter === null) {
+                throw new \Exception(intl_get_error_message(), intl_get_error_code());
+            }
+        } catch (\Exception $e) {
```
Comment (samdark) [inline framework/i18n/MessageFormatter.php]: Is it throwing exactly `\Exception` or more specific one?

Comment (subdee) [inline framework/i18n/MessageFormatter.php]: @samdark Hm, it catches \IntlException so maybe that is indeed better to use, I will amend this.
@SilverFire I'm on it.

**Comment by cebe** [comment]
IntlException does not exist in PHP5.4 and the fix also does not seem to work this way on HHVM.

**Comment by samdark** [comment]
OK. Then general exception should be used.

**Comment by cebe** [comment]
@SilverFire @samdark I could fix it, are you fine with assigning it to me?

**Comment by cebe** [comment]
Applied a slightly different fix, thanks!

**Pull request closed without merging**